Added date time search feature enhancement .

RajPratim21 opened the pull request:
@Treora  please reveiw my code regarding the commit, I have followd all the standard practises of redux of acrtions, reducers, store and tried to eliminate all the errors possible before submiting it. and also documented it and added required comments. issue  #26

RajPratim21 committed:
Date-time search integrated commit

RajPratim21 committed:
Merge https://github.com/WebMemex/webmemex-extension

Treora commented:
Great that you got this working solution. Having a working proof is only the first step though, next is to turn it into nice, readable, documented and maintainable code.

It looks like your code still uses the global variable workaround; did you perhaps forget to push the cleaned up approach you mentioned? There is also a huge amount of code debris, stuff that you probably changed while hacking but that should obviously be removed from the PR (even some parts of the date-picker docs appear to be copied into the readme?). Please have a look through [the diff](https://github.com/WebMemex/webmemex-extension/pull/66/files) yourself, and try to make sure that every line in there is there for a reason. I would actually consider to start cleanly by checking out master, and one by one re-add the pieces that are part of the final solution.

We'll have to also find a nice way to use the css for the date picker without having to include it in our repo. Possibly the build process just has to copy it from `node_modules/...` into `extension/overview`.

RajPratim21 commented:
Thank you @Treora  for the appreciation, I will do the required to make the code "nice, readable, documented and maintainable" . 

The Global variables are still required because I am using them to set the data at the reducer.js , once the date-pickers are clicked that event is caught by `mapTodispatch` and inside
 that `onStartDateChange: date ` function  
we set the Global variables with the current values 
` Myclass.Global_startDate =date;`
and these Global variables are then accesed at reducer.js to change  the  state.startview.  `return Object.assign({},state, {endDate : MyClass.Global_endDate})
 } `

I will clear all the code debris and these Reame stuff. And regarding css stuff currently for this patcch I am not focussing on  finding nicer way as that's out of scope of this patch. at least for now.

Treora commented on src/overview/components/Overview.jsx:
```diff
@@ -1,48 +1,98 @@
+  //used for updatind the selcted date.
+      onStartDateChange: date => {
+             Myclass.Global_startDate =date;
+             dispatch(actions.handleStartChange({current_Start_date: date}))
```
Here you pass the date with the name `current_Start_date`, while in the `handleStartChange` reducer you are looking for the argument `startDate`. If you would make these names match, I suppose that would fix the issue and you could remove the whole workaround with the global variable.

Treora replied:
In case it was not clear to you why the names have to match, read [this introduction](https://timkendrick.co.uk/articles/2016/03/31/named-es6-arguments.html) to named arguments in ES6.

RajPratim21 replied:
Thanks @Treora  for pointing out . it does work without the work around here.

Treora commented:
> regarding css stuff currently for this patcch I am not focussing on finding nicer way as that's out of scope of this patch. at least for now.

Sure, I may look into it later.

RajPratim21 commented on src/search/find-visits.js:
```diff
@@ -73,19 +74,48 @@ export function getLastVisits({
+    {   
+    //if startDate has been updated by user then it's updates else default value is used
+        startDate = MyClass.Global_startDate.format('x');
+
```
@Treora I am having trouble to get the states here . can you suggest some work around.

RajPratim21 committed:
Removed Global var

RajPratim21 committed:
Removed extra spaces

RajPratim21 reviewed (commented):
I have removed Global vars and also cleaned up the code debris and maintained the indentation style.,

RajPratim21 committed:
changes in babel.rc

RajPratim21 committed:
changes in babel.rc

RajPratim21 committed:
changes in pack..json

Treora commented:
Much better. It still looks like half the lines of the diff are there for no reason and should be removed, but the feature itself looks implemented sensibly now. For cleaning up the code, you could also start again from master and just add every line that is actually required for this code. Or if you feel you spent enough time on this already, I would totally understand; I can also do the final cleanup/rewrite myself, based on your approach.

RajPratim21 commented:
@Treora  Thank you for appreciating my work. It will be a big help if you can help with the cliean up as I am really interested in implementing the Natural language time search and wanted to focus on improvement of full text search and these stuffs as I plan to work on that. It would be very kind of you if you can do the clean up. I can later have a look on it and can learn how to cleanup as I am still naiva at it.
Thank you @Treora .

RajPratim21 committed:
removed some extra lines at package.json

RajPratim21 committed:
removed some extra lines

RajPratim21 committed:
Remove some lines

RajPratim21 committed:
removed one line

RajPratim21 committed:
line removals

RajPratim21 committed:
addded files to original repo

RajPratim21 committed:
addded files to original repo

RajPratim21 commented:
@Treora  I have added the required files to the original mster repo, and done the final code cleanup , I hope you can merge it now.

blackforestboi commented:
Hey @RajPratim21 

any reason why you changed 22k files? :)

RajPratim21 commented:
@oliversauter  , may be due to I added all the changes from a new reppo it caused the problem, I have deleted previous repo and clonned a new repo and added the required lines to it, it's working and test and code is clean and understandable , the PR is continued to PR  #74

blackforestboi commented:
Unfortunately now it's impossible to see what work you have done. :/ 
Please find a way to only see the changed files, maybe rebasing helps: https://git-scm.com/book/en/v2/Git-Branching-Rebasing

RajPratim21 commented:
please refer to PR #74 for further evaluation. and can close this PR.
